packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -49,24 +49,34 @@ const fixId = id => {
 }
 exports.fixId = fixId
 
-const fixIds = object =>
-  _.mapValues(object, (val, key) => {
-    if (key === `sys`) {
-      val = {
-        ...val,
-        id: fixId(val.id),
-        contentful_id: val.id,
-      }
-    }
+const fixIds = object => _fixIds(object)
 
-    if (_.isArray(val)) {
-      return _.toArray(fixIds(val))
-    }
-    if (_.isPlainObject(val)) {
-      return fixIds(val)
+const _fixIds = (object, front = new Set()) => {
```

Comment:
Could you add a description on what this actually does 😬.

Reply:
Sure :)